### PR TITLE
Added gradle to leverage application plugin, moved compose top level

### PR DESCRIPTION
**Pull request by ieugen**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -10,8 +10,10 @@ services:
       - 443:443
       - 465:465
       - 993:993
-      
+
   james:
+    build:
+      context: local
```
Comment (dleangen) [inline docker-compose.yaml:16]: For my information: what is this for?

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ext {
+    jamesVersion = '3.5.0-SNAPSHOT'
+}
+
+dependencies {
+
+    compile "org.apache.james:james-server-jpa-guice:${jamesVersion}"
+    compile "org.apache.james:james-server-cli:${jamesVersion}"
+
+}
```
Comment (dleangen) [inline build.gradle:32]: Is this intended to compile James? Does this mean that the gradle build file will be placed in the James project? Is that the idea?

File: docker-compose.yaml (hunk 1 above)
Comment (ieugen) [reply docker-compose.yaml:16]: You can instruct docker-compose to build your image as part of the docker-compose up. 
https://docs.docker.com/compose/compose-file/#build

Comment (ieugen) [reply docker-compose.yaml:16]: Also see `docker-compose --help` and `docker-compose up --help` for how to use that when you have a running service.

File: build.gradle (hunk 2 above)
Comment (ieugen) [reply build.gradle:32]: No, those dependencies. I think I should make them "implementation" and it should work the same. 

I'm abusing the syntax a bit. I need to tell gradle what my dependencies are.

**Pull request merged**